### PR TITLE
Add hash and public simple() to FQName

### PR DESCRIPTION
**Pull request by Claude-at-Instaclustr**
Added hash derivation and made FQName::simple() public.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -19,6 +19,7 @@ hex = "0.4.3"
 num = { version = "0.4.0", features = ["serde"] }
 uuid = { version = "0.8.2", features = ["serde"]}
 bigdecimal = {version ="0.3.0", features = ["serde"]}
+serde = { version = "1.0.111", features = ["derive"] }
```
Comment (conorbros) [inline Cargo.toml:23]: Why this specific version of `serde`? We can use the latest.

**Review by conorbros — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (rukai) [reply Cargo.toml:23]: It doesn't matter.
That will still evaluate to the latest 1.x.y release as it's still semver compatible

Comment (conorbros) [reply Cargo.toml:23]: Sure, but why specify this version if we don't need to? Anyhow not going to block on this as it's pretty pedantic.

Comment (Claude-at-Instaclustr) [reply Cargo.toml:23]: version copied from Shotover.

Comment (conorbros) [reply Cargo.toml:23]: No problems. Once you rebase we can get this merged.

Comment (conorbros) [reply Cargo.toml:23]: If you want to release this, open a PR changing the version in `Cargo.toml`. Then tag the commit when it's merged.

**Review by rukai — APPROVED** [review]
lgtm

**Pull request merged**